app/packages/core/src/components/Grid/useRecords.ts

```diff
@@ -0,0 +1,15 @@
+import type { MutableRefObject } from "react";
+import { useMemo, useRef } from "react";
+
+export type Records = MutableRefObject<Map<string, number>>;
+
+export default (clear: string) => {
+  const records = useRef(new Map<string, number>());
+
+  useMemo(() => {
+    clear;
+    records.current.clear();
+  }, [clear]);
```

Comment:
What's going on here? I don't get why it's a memo and not an effect.

Reply:
The spotlight instance is a memo, but I suppose an unmount in an effect makes more sense here 👍

Reply:
I think `useMemo` is the right call here. I removed the `useRef` weirdness. Now a new records `Map` is created with each new Spotlight instance. They have the same lifecycle, so I think this makes sense

Reply:
`return useMemo(() => new Map(), [clear])` makes sense to me. Just that previously it was being used like an effect and I couldn't wrap my head around it 😅